pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -228,6 +228,23 @@ stdenv.mkDerivation ({
         configureFlags+=" --extra-lib-dirs=$p/lib"
       fi
     done
+  '' + (optionalString stdenv.isDarwin ''
+    # Work around a limit in the Mac OS X Sierra linker on the number of paths
+    # referenced by any one dynamic library:
+    #
+    # Create a local directory with symlinks of the *.dylib (Mac OS X shared
+    # libraries) from all the dependencies.
+    local dynamicLinksDir="$out/lib/links"
+    mkdir -p $dynamicLinksDir
+    local foundDylib=false
```

Comment:
I don't see where `foundDylib` is used. Perhaps you wanted to set `foundDylib` to `"true"` in the `for` loop below, and then if no dylib is found, delete the (empty) `$out/lib/links` dir?